src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -395,14 +429,19 @@ private async Task RemoveContainerNetworkAsync(IExecutionContext executionContex
             }
         }
 
-        private async Task ContainerHealthcheck(IExecutionContext executionContext, ContainerInfo container)
+        private async Task<string> ContainerHealthcheck(IExecutionContext executionContext, ContainerInfo container)
         {
             string healthCheck = "--format=\"{{if .Config.Healthcheck}}{{print .State.Health.Status}}{{end}}\"";
             string serviceHealth = (await _dockerManager.DockerInspect(context: executionContext, dockerObject: container.ContainerId, options: healthCheck)).FirstOrDefault();
             if (string.IsNullOrEmpty(serviceHealth))
             {
+                string exitCode = "--format=\"{{print .State.ExitCode}}\"";
+                string serviceExitCode = (await _dockerManager.DockerInspect(context: executionContext, dockerObject: container.ContainerId, options: exitCode)).FirstOrDefault();
```

Comment:
why do we change how we detect health, feels like an out-of-scope change?

Reply:
Yes, it is a change made after last round of PR comments. I have decided to check also the container exit code. More context here: [doc](https://docs.google.com/document/d/1e5wAd1tZ8gM397Yes6LJ6OgmH1d8roFe06T6zPvesWE/edit)